src/main/com/diana/restaurant/queue/QueueManager.java

```diff
@@ -0,0 +1,58 @@
+package com.diana.restaurant.queue;
+
+import com.diana.restaurant.entities.EntityPriorityForQueue;
+import com.diana.restaurant.entities.Order;
+import com.diana.restaurant.enums.Queues;
+import com.diana.restaurant.exceptions.KeyNotFoundException;
+import com.diana.restaurant.exceptions.services.NullEntityException;
+import com.diana.restaurant.util.QueueManagerUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+public class QueueManager {
+  private static QueueManager instance;
+  private final Map<Queues, Queue> instanceMap = new HashMap<>();
+
+  private QueueManager() {
+    this.registerQueues(this.instanceMap);
+  }
+
+  private void registerQueues(Map<Queues, Queue> instanceMap) {
+    instanceMap.put(
+        Queues.CHECK_INGREDIENTS_QUEUE, new PriorityQueue<EntityPriorityForQueue<Order>>());
+  }
+
+  public static QueueManager getInstance() {
+    return Optional.ofNullable(instance).orElseGet(QueueManager::new);
+  }
+
+  private <T> T get(String key) {
+    Optional.ofNullable(key).orElseThrow(NullPointerException::new);
+    return (T)
+        Optional.ofNullable(instanceMap.get(key))
+            .orElseThrow(() -> new KeyNotFoundException("Queue"));
+  }
+
+  private void enQueue(String queueName, EntityPriorityForQueue entity) {
+    var queue =
+        (PriorityQueue)
+            Optional.ofNullable(this.get(queueName))
+                .orElseThrow(() -> new KeyNotFoundException("Queue"));
+    Optional.of(queue).filter(q -> q.size()<= QueueManagerUtil.MAX_QUEUE_CAPACITY).orElseThrow();
```

Comment:
I think max capacity should live within the Queue enum, feeel free move it there.

Reply:
solved